widget.subscribe.php

```diff
@@ -413,6 +415,38 @@ function add_subscriber() {
 				}
 			}
 
+            $recaptcha_token = sanitize_text_field( $_POST['captcha_token'] );
+
+			if ( ! empty( $sailthru['google_recaptcha_site_key'] ) && ! empty( $sailthru['google_recaptcha_secret'] ) && ! empty( $recaptcha_token ) ) {
+				write_log( "reCaptcha enabled, verifying" );
+				try {
+					$response = wp_remote_get( 'https://www.google.com/recaptcha/api/siteverify?secret=' . $sailthru['google_recaptcha_secret'] . '&response=' . $_POST['captcha_token'] );
+					$body = wp_remote_retrieve_body( $response );
+					$data = json_decode( $body, true );
+					if ( false == $data['success'] ) {
+						// failed, send an error message here, but keep it vague
+						$this->return_response(
+							array(
+								'success' => false,
+								'message' => 'Sorry, something went wrong and we could not subscribe you.'
+							)
+						);
+					}
+				} catch ( Exception $e ) {
+					write_log( $e->getMessage() );
+				}
+			}
+
+			$reset_optout_status = isset( $_POST['reset_optout_status'] ) && ! empty( $_POST['reset_optout_status'] ) ? 'none': '';
+
+			$profile_data = array(
+				'id'    => $email,
+				'key'   => 'email',
+				'vars'  => $options['vars'],
+				'lists' => $options['lists'],
+				'optout_email' => $reset_optout_status
+			);
```

Comment:
Non-blocking comment:
The whitespace layout of this code is using 2 styles, which can impact readability. I suggest using one of the following styles consistently here (and in the future across this repo).

Option 1: Common Spacing
```
$profile_data = array(
    'id' => $email,
    'key' => 'email',
    'vars' => $options['vars'],
    'lists' => $options['lists'],
    'optout_email' => $reset_optout_status
);
```

Option 2: Parallel arrows
```
$profile_data = array(
    'id'           => $email,
    'key'          => 'email',
    'vars'         => $options['vars'],
    'lists'        => $options['lists'],
    'optout_email' => $reset_optout_status
);
```